lib/league_stat.rb

```diff
@@ -0,0 +1,29 @@
+class LeagueStat
+
+  attr_reader :stats_by_team
+
+  def initialize(teams_list, games_list)
+    @stats_by_team = Hash.new do |hash, key|
+      hash[key] = Hash.new { |hash, key| hash[key] = 0 }
+    end
```

Comment:
this will look like {
:team_id (one for each team) => {
  :team_id => "name",
  :team_stat => 0, (default value so can += 1 for each game, goal scored, etc...
  }
and can then iterate through this hash with individual methods to generate the correct return values

Reply:
it didn't keep the spacing... it's a hash with default value 0 within a hash of teams, so each team can be iterated through, in case you couldn't read that first comment

Reply:
I really like this idea!